astropy/units/tests/test_format.py

```diff
@@ -583,29 +583,23 @@ def test_format_styles_non_default_fraction(format_spec, fraction, string, decom
     assert fluxunit.decompose().to_string(format_spec, fraction=fraction) == decomposed
 
 
-@pytest.mark.parametrize("format_spec", ["generic", "cds", "fits", "ogip", "vounit"])
-def test_no_multiline_fraction(format_spec):
+@pytest.mark.parametrize("format_spec", u_format.Base.registry.keys())
+def test_multiline_fraction_different_if_available(format_spec):
     fluxunit = u.W / u.m**2
-    with pytest.raises(
-        ValueError,
-        match=(
-            f"^'{format_spec}' format only supports 'inline' fractions, "
-            r"not fraction='multiline'\.$"
-        ),
-    ):
-        fluxunit.to_string(format_spec, fraction="multiline")
+    inline_format = fluxunit.to_string(format_spec, fraction="inline")
+    if format_spec in ["generic", "cds", "fits", "ogip", "vounit"]:
```

Comment:
What's the advantage of writing a single test with `if`- and `else`-blocks in it instead of two separate tests without such blocks? Is it to avoid writing out lists of formatter names?

Reply:
Mostly wrote it like this so that if we add a new format, we will automatically test it (and thus be warned if it does the wrong thing).